src/components/TextField/TextField.scss

```diff
@@ -0,0 +1,169 @@
+@import "../variables";
+
+$textFieldLabelColor: $charcoal !default;
+
+.uir-textfield {
+    position: relative;
+    display: inline-block;
+    font-family: $font-family;
+    font-size: 13px;
+    font-weight: 400;
+    color: $textFieldLabelColor;
+
+    &:after {
+        position: absolute;
+        bottom: 0;
+        left: 0;
+        display: block;
+        width: 40%;
+        height: 1px;
+        content: "";
+        background: linear-gradient(to right, $grey 0%, transparent 100%);
+        transition: width .25s;
+    }
+
+    &--focus:after {
+        width: 100%;
+    }
+
+    &--invalid:after {
+        background: linear-gradient(to right, $pink 0%, transparent 100%);
+    }
+
+    // @NB: removed at Antoine's request, this may be restored later
+    // &--valid:after {
+    //     background: linear-gradient(to right, $teal 0%, transparent 100%);
+    // }
```

Comment:
Same as my comment on the TextArea, I think we should remove this until it's needed.

Reply:
Removed, same as TextArea